lib/hawkular/inventory/inventory_api.rb

```diff
@@ -255,178 +171,60 @@ def list_metrics_for_resource_type(resource_type_path)
-      end
-      Resource.new(res)
-    end
-
-    # Create a new metric type for a feed
-    # @param [String] feed_id Id of the feed
-    # @param [String] metric_type_id Id of the metric type to create
-    # @param [String] type Type of the Metric. Allowed are GAUGE,COUNTER, AVAILABILITY
-    # @param [String] unit Unit of the metric
-    # @param [Numeric] collection_interval
-    # @return [MetricType] Type just created or the one from the server if it already existed.
-    def create_metric_type(feed_id, metric_type_id, type = 'GAUGE', unit = 'NONE', collection_interval = 60)
-      the_feed = hawk_escape_id feed_id
-
-      metric_kind = type.nil? ? 'GAUGE' : type.upcase
-      fail "Unknown type #{metric_kind}" unless %w(GAUGE COUNTER AVAILABILITY').include?(metric_kind)
-
-      mt = build_metric_type_hash(collection_interval, metric_kind, metric_type_id, unit)
-
-      begin
-        http_post("/entity/f;#{the_feed}/metricType", mt)
-      rescue HawkularException => error
-        # 409 We already exist -> that is ok
-        raise unless error.status_code == 409
+      raw_hash = get_raw_entity_hash(path)
+      unless raw_hash
+        exception = HawkularException.new("Resource not found: #{resource_path}")
+        fail exception
       end
-
-      new_mt = http_get("/entity/f;#{the_feed}/mt;#{metric_type_id}")
-
-      MetricType.new(new_mt)
+      entity_hash = entity_json_to_hash(-> (_) { path }, raw_hash, fetch_properties)
+      Resource.new(entity_hash)
     end
 
     # List operation definitions (types) for a given resource type
     # @param [String] resource_type_path canonical path of the resource type entity
     # @return [Array<String>] List of operation type ids
     def list_operation_definitions(resource_type_path)
-      parsed_path = CanonicalPath.parse(resource_type_path.to_s)
-      feed_id = parsed_path.feed_id
-      resource_type_id = parsed_path.resource_type_id
-      ots = http_get("/traversal/f;#{feed_id}/rt;#{resource_type_id}/type=ot")
+      path = resource_type_path.is_a?(CanonicalPath) ? resource_type_path : CanonicalPath.parse(resource_type_path)
+      fail 'Missing feed_id in resource_type_path' unless path.feed_id
+      fail 'Missing resource_type_id in resource_type_path' unless path.resource_type_id
+      response = http_post(
+        '/strings/raw/query',
+        fromEarliest: true,
+        order: 'DESC',
+        tags: path.to_tags)
+      structures = extract_structures_from_body(response)
       res = {}
-      ots.each do |ot|
-        ot_name = ERB::Util.url_encode ot['name']
-        pts = http_get("/traversal/f;#{feed_id}/rt;#{resource_type_id}/ot;#{ot_name}/d;parameterTypes")
-        ot['parameters'] = pts[0]['value'] unless pts.empty?
-        od = OperationDefinition.new ot
-        res.store od.name, od
+      structures.map { |rt| rt['inventoryStructure'] }
+        .find_all { |rt| (rt.key? 'children') && (rt['children'].key? 'operationType') }
```

Comment:
is "select" preferable than "find_all"? It seems to be exactly the same in this case (Array)